lib/services/styles.js

```diff
@@ -0,0 +1,220 @@
+
+  return this.client({
+    path: constants.API_STYLES_DELETE,
+    params: {
+      owner: this.owner,
+      style: style
+    },
+    method: 'delete',
+    callback: callback
+  }).entity();
+};
+
+/**
+ * Reads a particular style.
+ *
+ * @param {string} style the id for an existing style
+ * @param {Function} callback called with (err)
+ * @returns {undefined} nothing, calls callback
+ * @example
+ * var MapboxClient = require('mapbox');
+ * var client = new MapboxClient('ACCESSTOKEN');
+ * client.deleteStyle('style-id', function(err) {
+ *   if (!err) console.log('deleted!');
+ * });
+ */
+Styles.prototype.readStyle = function(style, callback) {
+  invariant(typeof style === 'string', 'style must be a string');
+
+  return this.client({
+    path: constants.API_STYLES_READ,
+    params: {
+      owner: this.owner,
+      style: style
+    },
+    callback: callback
+  }).entity();
+};
+
+/**
+ * Read sprite
+ *
+ * @param {string} style the id for an existing style
+ * @param {Object={}} options
+ * @param {boolean} options.retina whether the sprite JSON should be for a
+ * retina sprite.
+ * @param {Function} callback called with (err)
+ * @returns {undefined} nothing, calls callback
+ * @example
+ * var MapboxClient = require('mapbox');
+ * var client = new MapboxClient('ACCESSTOKEN');
+ * client.readSprite('style-id', {
+ *   retina: true
+ * }, function(err) {
+ *   if (!err) console.log('deleted!');
+ * });
+ */
+Styles.prototype.readSprite = function(style, options, callback) {
+  invariant(typeof style === 'string', 'style must be a string');
+
+  if (typeof options === 'function') {
+    callback = options;
+    options = {};
+  }
+
+  var retina = '';
+  if (options.retina) {
+    invariant(typeof options.retina === boolean,
+      'retina option must be a boolean value');
+    if (options.retina) {
+      retina = '@2x';
+    }
+  }
+
+  var format = 'json';
+  if (options.format) {
+    invariant(options.format === 'json' ||
+      options.format === 'png',
+      'format parameter must be either json or png');
+    format = options.format;
+  }
+
+  return this.client({
+    path: constants.API_STYLES_SPRITE,
+    params: {
+      owner: this.owner,
+      retina: retina,
+      format: format,
+      style: style
+    },
+    callback: callback
+  }).entity();
+};
+
+/**
+ * Add an icon to a sprite.
+ *
+ * @param {string} style the id for an existing style
+ * @param {string} iconName icon's name
+ * @param {Buffer} icon icon content as a buffer
+ * @param {Function} callback called with (err)
+ * @returns {undefined} nothing, calls callback
+ * @example
+ * var MapboxClient = require('mapbox');
+ * var fs = require('fs');
+ * var client = new MapboxClient('ACCESSTOKEN');
+ * client.addIcon('style-id', fs.readFileSync('icon.png'), function(err) {
+ *   if (!err) console.log('added icon!');
+ * });
+ */
+Styles.prototype.addIcon = function(style, iconName, icon, callback) {
```

Comment:
Because there isn't a custom `Content-Type` header, the mime interceptor is trying to employ the default, in this case `application/json` as [configured here](https://github.com/mapbox/mapbox-sdk-js/blob/master/lib/client.js#L11). Add this to the request:

```
headers: {
  'Content-Type': 'application/octet-stream'
}
```